packages/ui/src/components/typeahead/Typeahead.tsx

```diff
@@ -22,21 +22,36 @@ export const Typeahead: FunctionComponent<TypeaheadProps> = ({
   onChange,
   onCleanInput,
   'data-testid': dataTestId,
+  onCreateNewBean,
 }) => {
+  const createNewBeanValue = 'create-new-bean';
+  const createNewBeanWithNameValue = 'create-new-bean-with-name';
   const [inputValue, setInputValue] = useState<string>(selectedItem?.name ?? '');
   const [isOpen, setIsOpen] = useState(false);
   const inputRef = useRef<HTMLInputElement>(null);
 
   const items = useMemo(() => {
+    if (onCreateNewBean) {
+      itemsProps = [...itemsProps, { name: 'Create new bean', description: '', value: createNewBeanValue }];
+    }
+    setInputValue(selectedItem?.name ?? '');
     if (itemsProps.find((item) => item.name === selectedItem?.name) || selectedItem === undefined) {
       return itemsProps;
     }
-    return [...itemsProps, { name: selectedItem?.name, value: selectedItem?.value }];
-  }, [itemsProps, selectedItem]);
+    return [{ name: selectedItem?.name, value: selectedItem?.value }, ...itemsProps];
+  }, [itemsProps, selectedItem, onCreateNewBean, createNewBeanValue]);
 
   const onItemChanged = useCallback(
     (_event: unknown, name: string | number | undefined) => {
-      if (!isDefined(name)) {
+      if (name === createNewBeanValue) {
+        onCreateNewBean?.(createNewBeanValue);
+        setIsOpen(false);
+        return;
+      } else if (name === createNewBeanWithNameValue) {
+        onCreateNewBean?.(createNewBeanWithNameValue, inputValue);
+        setIsOpen(false);
+        return;
```

Comment:
Could this section be repurposed to behave the same regardless of whether we have a value or not?

Reply:
We have the `onCreateNewBean` callback and we call it with some parameters, what would be the difference between calling it always with the `inputValue`, even if it is undefined?

What I wonder is if we could consolidate that code to say:
> Create a new Bean with the following values: `{ name: inputValue, type: javaType }`
>
Then, those values are applied to the beans model, if they are set, they will be shown, otherwise they will be `undefined`